services/121-service/test/program/delete-program.test.ts

```diff
@@ -0,0 +1,78 @@
+import { HttpStatus } from '@nestjs/common';
+
+import { SeedScript } from '@121-service/src/scripts/seed-script.enum';
+import {
+  deleteProgram,
+  getProgram,
+  startCbeValidationProcess,
+} from '@121-service/test/helpers/program.helper';
+import { seedPaidRegistrations } from '@121-service/test/helpers/registration.helper';
+import {
+  getAccessToken,
+  resetDB,
+} from '@121-service/test/helpers/utility.helper';
+import {
+  programIdOCW,
+  programIdPV,
+  registrationCbe,
+  registrationsOCW,
+  registrationsPV,
+} from '@121-service/test/registrations/pagination/pagination-data';
+
+describe('Delete program', () => {
+  let accessToken: string;
+
+  it('should delete a nlrc programs', async () => {
+    // Arrange
+    await resetDB(SeedScript.nlrcMultiple);
+    accessToken = await getAccessToken();
+
+    // Create some test data which should be cascaded deleted
+    await seedPaidRegistrations(registrationsOCW, programIdOCW);
+    await seedPaidRegistrations(registrationsPV, programIdPV);
+
+    // Act + Assert
+    const secretDto = { secret: process.env.RESET_SECRET! };
+
+    const deleteResponseOCW = await deleteProgram(
+      programIdOCW,
+      accessToken,
+      secretDto,
+    );
+    expect(deleteResponseOCW.statusCode).toBe(HttpStatus.NO_CONTENT);
+
+    const deleteResponsePV = await deleteProgram(
+      programIdPV,
+      accessToken,
+      secretDto,
+    );
+    expect(deleteResponsePV.statusCode).toBe(HttpStatus.NO_CONTENT);
+
+    const getProgramResponseOCW = await getProgram(programIdOCW, accessToken);
+    expect(getProgramResponseOCW.statusCode).toBe(HttpStatus.NOT_FOUND);
+
+    const getProgramResponsePV = await getProgram(programIdPV, accessToken);
+    expect(getProgramResponsePV.statusCode).toBe(HttpStatus.NOT_FOUND);
+  });
+
+  it('should delete eth programs', async () => {
```

Comment:
why ETH, and not also others (KRCS with safaricom-transfer.entity). (I saw your comment elsewhere about the eslint rule already catching a lot)

Reply:
I started with ETH and than I started to have doubts about the usefulness of doing it for all Fsps/ programs. But ETH was already done, so I left it..

Reply:
Ok I guess it's fine indeed. With the eslint present.